server/src/main/java/org/opensearch/cluster/health/ClusterStateHealth.java

```diff
@@ -73,7 +73,7 @@ public final class ClusterStateHealth implements Iterable<ClusterIndexHealth>, W
      * @param clusterState The current cluster state. Must not be null.
      */
     public ClusterStateHealth(final ClusterState clusterState) {
-        this(clusterState, clusterState.metadata().getConcreteAllIndices());
+        this(clusterState, clusterState.metadata().getConcreteAllIndices().toArray(String[]::new));
```

Comment:
Same comment as above- Does it make sense to change `concreteIndices` to lists as well? Instead of doing a back and forth to arrays?

Reply:
Addressed it in new PR https://github.com/opensearch-project/OpenSearch/pull/14723 (This PR has screwed-up after multiple rebase)